src/main/java/org/edumips64/core/Converter.java

```diff
@@ -39,57 +39,32 @@
    * @throws IrregularStringOfBitsException if the string of bits is not well-formed or the string of bits contains
    * a value that cannot be stored into an <code>int</code> variable
    */
+
   public static int binToInt(String bits, boolean unsignd) throws IrregularStringOfBitsException {
-    if (bits.length() > 32) {
-      throw new IrregularStringOfBitsException();
-    }
 
-    if (unsignd && bits.length() == 32 && bits.charAt(0) == '1') {
+    //check for string irregularities to begin with
+    if ((bits.length() > 32) || (unsignd && bits.length() == 32 && bits.charAt(0) == '1') || !isBinaryString(bits)) {
       throw new IrregularStringOfBitsException();
     }
 
-
     //se la stringa di bits Ãš lunga 32 bit
     //ed Ãš composta da uno 1 e 31 0
     //allora la conversione si deve fare a mano
     //perchÃš il numero da ritornare Ãš -(2^32)
     //e non si puÃ² utilizzare il valore positivo (2^32)
     //con il tipo int :-(
-    if (!unsignd && bits.length() == 32 && bits.charAt(0) == '1') {
-      boolean overflow = true;
-
-      for (int i = 1; i < bits.length(); i++) {
-        if (bits.charAt(i) != '0') {
-          overflow = false;
-          break;
-        }
-      }
-
-      if (overflow) {
-        return (int)(-Math.pow(2.0, 32.0));
-      }
+    if (!unsignd && bits.length() == 32 && isOverflow(bits)) {
+      return (int)(-Math.pow(2.0, 32.0));
```

Comment:
Can you please store this value in a constant? It's more efficient than computing it every time.

Also, as a general point, if you want to do this type of operations (working with powers of two) it's better to use shift operators rather than using floating-point operations and then converting the result to int.